src/sentry/integrations/vsts/client.py

```diff
@@ -199,9 +201,19 @@ def _refresh_auth_if_expired(self):
         if time_expires is None:
             raise InvalidIdentity("VstsApiClient requires identity with specified expired time")
         if int(time_expires) <= int(time()):
-            self.identity.get_provider().refresh_identity(
-                self.identity, redirect_url=self.oauth_redirect_url
-            )
+            # TODO(iamrajjoshi): Remove this after migration
+            # Need this here because there is no way to get any identifier which would tell us which method we should use to refresh the token
+            from sentry.identity.vsts.provider import VSTSNewIdentityProvider
+
+            try:
+                self.identity.get_provider().refresh_identity(
```

Comment:
does the integration need to migrated for us to be able to use the new refresh method? when does this fail?

Reply:
yep new integration needs the new method. this method will fail when we try to use the old refresh method on the new integration, then it will try using the new method.

Reply:
could you find the `Integration` from the `org_integration_id` and check the metadata?

Reply:
hmm, thats a good point, let me try